src/common/components/repository-row/index.js

```diff
@@ -149,9 +149,10 @@ class RepositoryRow extends Component {
   componentWillReceiveProps(nextProps) {
     if (nextProps.registerNameStatus.success &&
         !this.props.registerNameStatus.success) {
-      this.closeUnregisteredTimerID = window.setTimeout(
-        this.closeUnregisteredDropdown.bind(this), 2000
-      );
+      this.closeUnregisteredTimerID = window.setTimeout(() => {
+        this.closeUnregisteredDropdown();
+        this.props.dispatch(registerNameClear(this.props.fullName));
```

Comment:
We need to clear state of register name API calls, because user may open up the dropdown again to change the name (and shouldn't see 'success' status there).